hyperopt/base.py

```diff
@@ -226,6 +228,18 @@ def spec_from_misc(misc):
     return spec
 
 
+def validate_timeout(timeout):
+    if timeout is not None and (
+            not isinstance(timeout, numbers.Number)
+            or timeout <= 0
+            or isinstance(timeout, bool)
```

Comment:
I think this is redundant. If it's not a number, you don't need to check if it's a bool

Reply:
also, if the docstring says int, then I think better to check here is int, rather than any number

Reply:
This is not my code segment, I used check from spark implementation, but since I use the same parameter I just moved this check to separate method to base file. To not break something by accident I did not change the logic of this validation.

Reply:
1) if you can reuse code from somwhere else, do it. Do not duplicate
2) Either the docstring or check need to be changed, to match. The check allows for floats and has a check for bool that seems useless to me, while the docstring states int. I would change this code as I don't think we need float precision for a timeout in seconds

Reply:
1. It was reused, not duplicated. As I mention, this check was just moved to base.
2. I think this should be discussed when this check was introduced to spark implementation. Again, I prefer to reuse it than make (almost) the same new check.